client/sites/components/sites-dashboard.tsx

```diff
@@ -165,7 +165,9 @@ const SitesDashboard = ( {
 		sitesFilterCallback,
 		'all',
 		[ 'is_a4a_dev_site', 'site_migration' ],
-		[ 'theme_slug' ]
+		[ 'theme_slug' ],
+		// Don't fetch sites on narrow screens since it's not visible.
+		! selectedSite || isWide
```

Comment:
This seems wrong. We utilize `allSites` in many places across this component and it might be unclear as to why the data doesn't appear for mobile devices. If we're doing things like this, it might mean we need to rearchitect things a little bit so the queries and hooks are called only where necessary. It might mean further componentization needs to happen.